moebot_bot/bot/commands/channelRotationScheduler.go

```diff
@@ -0,0 +1,206 @@
+}
+
+func (s *ChannelRotationScheduler) Help() string {
+	return "`" + s.Keyword() + " -channels <channels> -interval <interval>`: Rotates through the specified channels, making them visible one at a time. \r\n " +
+		"The command doesn't edit already existing permissions, so when using this command make sure that only the first channel in the list is currently visible."
+}
+
+func (s *ChannelRotationScheduler) AddScheduledOperation(comm *CommPackage) error {
+	params := ParseCommand(comm.params, []string{"-channels", "-interval"})
+	if params["-channels"] == "" {
+		comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, you need to specify a list of channels.")
+		return fmt.Errorf("-channels parameter empty")
+	}
+	if params["-interval"] == "" {
+		comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, you need to specify a time interval.")
+		return fmt.Errorf("-interval parameter empty")
+	}
+
+	intervalString, err := parseCommandInterval(params["-interval"])
+	if err != nil {
+		comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, the interval you specified is invalid. You need to specify the interval in the format `XWXDXh`, for example `5W6D4h` for 5 weeks, 6 days and 4 hours.")
+		return err
+	}
+
+	server, err := db.ServerQueryOrInsert(comm.guild.ID)
+	if err != nil {
+		comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, there was a problem retrieving the current server informations. Please try again.")
+		return err
+	}
+	channels := []string{}
+	for _, c := range strings.Split(params["-channels"], " ") {
+		channels = append(channels, strings.Trim(c, "<#>"))
+	}
+	for _, chUID := range channels {
+		ch, err := comm.session.Channel(chUID)
+		if err != nil {
+			comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, there was a problem retrieving channel informations. Please try again.")
+			return err
+		}
+		if ch.GuildID != comm.guild.ID {
+			comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, there was a problem retrieving channel informations. Please try again.")
+			return err
+		}
+	}
+
+	err = db.ChannelRotationAdd(server.Id, channels[0], channels, intervalString)
+	if err != nil {
+		comm.session.ChannelMessageSend(comm.channel.ID, "Sorry, there was a problem adding the rotation to the server.")
+		return err
+	}
+	comm.session.ChannelMessageSend(comm.channel.ID, "Channel rotation successfully added")
+	return nil
+}
+
+func (s *ChannelRotationScheduler) OperationDescription(operationID int64) string {
+	channelRotation, err := db.ChannelRotationQuery(operationID)
+	if err != nil {
+		return "Failed to retrieve channel list"
+	}
+	result := "Rotating channels"
+	for _, c := range channelRotation.ChannelUIDList {
+		result += " <#" + c + ">"
+	}
+	return result
+}
+
+func parseCommandInterval(interval string) (string, error) {
+	intervalsOrder := []string{"Y", "M", "W", "D", "h", "m"}
+	rx, _ := regexp.Compile("^(\\d+[YMWDhm]){1}(\\d+[YMWDhm]){0,1}(\\d+[YMWDhm]){0,1}(\\d+[YMWDhm]){0,1}(\\d+[YMWDhm]){0,1}$")
```

Comment:
This is very worthy of a comment explaining what the regex is trying to accomplish as it's not immediately clear without reading all the rest of the code